iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -514,3 +559,102 @@ def dump_streamers(self):
         """Dump all of the streamers in this sensor graph as a list of strings."""
 
         return [str(streamer) for streamer in self.streamers]
+
+    def dump_constant_database(self):
+        """Dump all of the constants in this sensor graph as a list of strings."""
+        
+        constant_dump = []
+
+        for stream, value in sorted(self.constant_database.items(), key=lambda x: x[0].encode()):
+            constant_dump.append("'{}' {}".format(stream, value))
+
+        return constant_dump
+
+    def dump_metadata_database(self):
+        """Dump all of the metadata in this sensor graph as a list of strings."""
+
+        metadata_dump = []
+
+        for metadata, value in sorted(self.metadata_database.items(), key=lambda x: x[0].encode()):
+            metadata_dump.append("{}: {}".format(metadata, value))
+
+        return metadata_dump
+
+    def dump_config_database(self, dump_config_type=True, ignore_configs=[]):
+        """Dump all of the config variables in this sensor graph as a list of strings."""
+
+        config_dump = []
+        for slot, conf_vars in self.config_database.items():
+            for conf_var, conf_def in conf_vars.items():
+                conf_type, conf_val = conf_def
+
+                if conf_var in ignore_configs:
+                    continue
```

Comment:
We typically want to ignore the actual config variable that holds our checksum calculation when we are doing the checksum calculation. If we don't ignore this config variable, we will have a chicken and egg problem when trying to figure out the checksum.